### PR TITLE
Fix some build warnings, allow project to use latest C# language version

### PR DESCRIPTION
**Pull request by MisterZeus**
The Debug config of the project was hard-coded to use C#5 for some reason.
That is what was breaking my Appveyor builds when I tried following any of Visual Studio's recommendations for code cleanup.

This PR allows the Debug config to use the latest C# lang,
upgrades the project from .NET 4.6.1 to 4.7.2 and fixes a few of the warnings encountered during the Release Build.

A side effect of this upgrade is that now the Debug Build is picking up a lot more subtle warnings, while still building correctly. Here is an example: https://ci.appveyor.com/project/MisterZeus/sqlquerystress/builds/32269642

### CODE REVIEW

**Comment by MisterZeus** [comment]
Hmm, your Appveyor build uses a different version of Microsoft's build engine,
so these upgrades are not working for you, but they are for me...
Shall I close this PR and leave the changes just in my branch?
Or is there some easy fix I'm not seeing?

File: src/SQLQueryStress/App.config [hunk 1]
```diff
@@ -3,4 +3,4 @@
   <runtime>
     <gcServer enabled="true"/>
   </runtime>
-<startup><supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.6.1"/></startup></configuration>
+<startup><supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.7.2"/></startup></configuration>
```
Comment (ErikEJ) [inline src/SQLQueryStress/App.config:6]: Can we revert this, the tool is most likely used on many systems that do not have  .NET 4.7.2 installed yet

File: src/SQLQueryStress/Form1.cs [hunk 2]
```diff
@@ -472,13 +472,25 @@ private void UpdateUi()
 
         private void btnCleanBuffer_Click(object sender, EventArgs e)
         {
+            if (!_settings.MainDbConnectionInfo.TestConnection())
```
Comment (ErikEJ) [inline src/SQLQueryStress/Form1.cs:475]: Did you forget to rebase on the latest master?

File: src/SQLQueryStress/Form1.cs [hunk 3]
```diff
@@ -472,13 +472,25 @@ private void UpdateUi()
...
+            }
+
             MessageBox.Show(LoadEngine.ExecuteCommand(_settings.MainDbConnectionInfo.ConnectionString, "DBCC DROPCLEANBUFFERS")
                 ? "Buffers cleared"
                 : "Errors encountered");
         }
 
         private void btnFreeCache_Click(object sender, EventArgs e)
         {
+            if (!_settings.MainDbConnectionInfo.TestConnection())
```
Comment (ErikEJ) [inline src/SQLQueryStress/Form1.cs:488]: Already in master branch

File: src/SQLQueryStress/SQLQueryStress.csproj [hunk 4]
```diff
@@ -19,7 +19,7 @@
     <ManifestKeyFile>SQLQueryStress_TemporaryKey.pfx</ManifestKeyFile>
     <IsWebBootstrapper>false</IsWebBootstrapper>
     <ApplicationIcon>Resources\QueryStressIcon.ico</ApplicationIcon>
-    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
```
Comment (ErikEJ) [inline src/SQLQueryStress/SQLQueryStress.csproj:22]: Can we revert?

**Comment by ErikEJ** [comment]
I updated the CI (was building on VS 2015, now updated to 2019

File: src/SQLQueryStress/App.config (hunk 1 above)
Comment (MisterZeus) [reply src/SQLQueryStress/App.config:6]: Yup, I have reverted this and rebased my fix-warnings branch on my master too.
I committed the change a few minutes ago, but Github has not updated yet?

File: src/SQLQueryStress/Form1.cs (hunk 2 above)
Comment (MisterZeus) [reply src/SQLQueryStress/Form1.cs:475]: Sorry, I didn't realise I needed to do a rebase, learning as I go. Rebased my fix-warnings branch on my master now.

File: src/SQLQueryStress/Form1.cs (hunk 3 above)
Comment (MisterZeus) [reply src/SQLQueryStress/Form1.cs:488]: Rebased my fix-warnings branch on my master now.

File: src/SQLQueryStress/SQLQueryStress.csproj (hunk 4 above)
Comment (MisterZeus) [reply src/SQLQueryStress/SQLQueryStress.csproj:22]: Reverted back to 4.6.1

**Comment by MisterZeus** [comment]
Thank you got upgrading it to 2019!

File: src/SQLQueryStress/App.config (hunk 1 above)
Comment (MisterZeus) [reply src/SQLQueryStress/App.config]: aha, I think I've done it correctly now.

**Pull request merged**